dbal/dsn_test.go

```diff
@@ -1,25 +1,38 @@
 package dbal
 
 import (
-	"testing"
-
 	"github.com/stretchr/testify/require"
+	"testing"
 )
 
 func TestIsMemorySQLite(t *testing.T) {
-	require.True(t, IsMemorySQLite(SQLiteInMemory))
-	require.True(t, IsMemorySQLite(SQLiteSharedInMemory))
-	require.True(t, IsMemorySQLite("sqlite://file:uniquedb:?_fk=true&mode=memory"))
-	require.True(t, IsMemorySQLite("sqlite://file:uniquedb:?_fk=true&cache=shared"))
-	require.True(t, IsMemorySQLite("sqlite://file:uniquedb:?_fk=true&mode=memory&cache=shared"))
-	require.True(t, IsMemorySQLite("sqlite://file:uniquedb:?_fk=true&cache=shared&mode=memory"))
-	require.False(t, IsMemorySQLite("sqlite://file:::uniquedb:?_fk=true&mode=memory"))
-	require.False(t, IsMemorySQLite("sqlite://"))
-	require.False(t, IsMemorySQLite("sqlite://file"))
-	require.False(t, IsMemorySQLite("sqlite://file:::"))
-	require.False(t, IsMemorySQLite("sqlite://?_fk=true&mode=memory"))
-	require.False(t, IsMemorySQLite("sqlite://?_fk=true&cache=shared"))
-	require.False(t, IsMemorySQLite("sqlite://file::?_fk=true"))
-	require.False(t, IsMemorySQLite("sqlite://file:::?_fk=true"))
-	require.False(t, IsMemorySQLite("postgresql://username:secret@localhost:5432/database"))
-}
+	testCases := map[string]bool{
+		SQLiteInMemory:               true,
+		SQLiteSharedInMemory:         true,
+		"memory":                     true,
+		"sqlite://:memory:?_fk=true": true,
+		"sqlite://file:uniquedb:?_fk=true&mode=memory":              true,
+		"sqlite://file:uniquedb:?_fk=true&cache=shared":             true,
+		"sqlite://file:uniquedb:?_fk=true&mode=memory&cache=shared": true,
+		"sqlite://file:uniquedb:?_fk=true&cache=shared&mode=memory": true,
+		"sqlite://file:::uniquedb:?_fk=true&mode=memory":            false,
+		"sqlite://":                                            false,
+		"sqlite://file":                                        false,
+		"sqlite://file:::":                                     false,
+		"sqlite://?_fk=true&mode=memory":                       false,
+		"sqlite://?_fk=true&cache=shared":                      false,
+		"sqlite://file::?_fk=true":                             false,
+		"sqlite://file:::?_fk=true":                            false,
+		"postgresql://username:secret@localhost:5432/database": false,
+	}
+
+	for k, v := range testCases {
+		t.Logf("testing: %s\nexpected:%v\n", k, v)
+		ok := IsMemorySQLite(k)
+		if v {
+			require.True(t, ok)
+		} else {
+			require.False(t, ok)
+		}
```

Comment:
I would say this is more of an `assert` because the test can still continue with the other cases if one fails.
`require` should be used for stuff that is required to continue, e.g. there could be panics otherwise or it is in some other broken state.
```suggestion
		assert.Equal(t, v, ok)
```

Reply:
Using require.Equal(t, v, IsMemorySQLite(k)) rather so that it fails and exits at the point of failure (for test debugging reasons) since this is in a loop.

Reply:
For debugging you can use the `-failfast` flag when executing tests.